pages/wards/visits/[id]/edit.js

```diff
@@ -0,0 +1,83 @@
+import React, { useState } from "react";
+import { GridRow, GridColumn } from "../../../../src/components/Grid";
+import Layout from "../../../../src/components/Layout";
+import ErrorSummary from "../../../../src/components/ErrorSummary";
+import VisitForm from "../../../../src/components/VisitForm";
+import verifyToken from "../../../../src/usecases/verifyToken";
+import Router from "next/router";
+import propsWithContainer from "../../../../src/middleware/propsWithContainer";
+import { WARD_STAFF } from "../../../../src/helpers/userTypes";
+
+const EditAVisit = ({
+  callId,
+  initialPatientName,
+  initialContactName,
+  initialContactNumber,
+  initialContactEmail,
+  initialCallDateTime,
+}) => {
+  const [errors, setErrors] = useState([]);
+
+  const submit = () => {
+    Router.push({
+      pathname: `/wards/visits/${callId}/edit-success`,
+    });
+  };
+
+  return (
+    <Layout
+      title="Edit a virtual visit"
+      hasErrors={errors.length != 0}
+      showNavigationBarForType={WARD_STAFF}
+      showNavigationBar={true}
+    >
+      <GridRow>
+        <GridColumn width="two-thirds">
+          <ErrorSummary errors={errors} />
+          <VisitForm
+            heading="Edit a virtual visit"
+            initialPatientName={initialPatientName}
+            initialContactName={initialContactName}
+            initialContactNumber={initialContactNumber}
+            initialContactEmail={initialContactEmail}
+            initialCallDateTime={initialCallDateTime}
+            errors={errors}
+            setErrors={setErrors}
+            submit={submit}
+          />
+        </GridColumn>
+      </GridRow>
+    </Layout>
+  );
+};
+export const getServerSideProps = propsWithContainer(
+  verifyToken(async ({ query, container }) => {
+    const callId = query.id;
+    const retrieveVisitByCallId = container.getRetrieveVisitByCallId();
```

Comment:
The `getRetrieveVisitByCallId` usecase is used in the patient / key contact visits pages to obfuscate the visit ID from the user. 

But here we're logged in as a ward staff so I feel the `[id]` in the path here should relate to the primary ID of the visit instead of the callID. What do you think?

Reply:
Makes sense!

Looks to me we'd have to do a couple of things if we want to do this including adding a new usecase to retrieve a visit by the `id` and maybe change our update a visit usecase to do it by `id` instead of `callId` to be consistent?

Reply:
Could I do dis in separate PRs after merging this one in?

Reply:
yep agreed